x86/sm2/bignum_montsqr_sm2.S

```diff
@@ -61,23 +62,24 @@
         sbb     d0, rbx
 
 S2N_BN_SYMBOL(bignum_montsqr_sm2):
+        CFI_START
         _CET_ENDBR
 
 #if WINDOWS_ABI
-        push    rdi
-        push    rsi
+        CFI_PUSH(rdi)
+        CFI_PUSH(rsi)
         mov     rdi, rcx
         mov     rsi, rdx
 #endif
 
 // Save more registers to play with
 
-        push    rbx
-        push    rbp
-        push    r12
-        push    r13
-        push    r14
-        push    r15
```

Comment:
I noticed much of the SM2 code uses `rbp` as a GP register whereas almost none of the other code seems to? I didn't look at the SM2 code but I suspect that wherever RBP is changed, more work needs to be done.